KAFKA-14188; Getting started for Kafka with KRaft

jsancio opened the pull request:
Update the quickstart HTML page to include how to quickly start and experiment with a Kafka cluster using KRaft instead of ZooKeeper.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

jsancio committed:
KAFKA-14188; Getting started for Kafka with KRaft

Update the quickstart HTML page to include how to quickly start and
experiment with a Kafka cluster using KRaft instead of ZooKeeper.

showuon commented on docs/quickstart.html:
```diff
@@ -46,6 +46,14 @@ <h4 class="anchor-heading">
+
         <p>
             Run the following commands in order to start all services in the correct order:
         </p>
```
Should we also remove the comment in zookeeper start command(L62)?
```
<pre class="line-numbers"><code class="language-bash"># Start the ZooKeeper service
[# Note: Soon, ZooKeeper will no longer be required by Apache Kafka.] <== remove this since it already happened
$ bin/zookeeper-server-start.sh config/zookeeper.properties</code></pre>
```

jsancio replied:
Done.

showuon commented on docs/quickstart.html:
```diff
@@ -64,6 +72,28 @@ <h4 class="anchor-heading">
+            Generate a Cluster UUID
+        </p>
+
+        <pre class="line-numbers"><code class="language-bash">$ KAFKA_CLUSTER_ID = $(bin/kafka-storege.sh random-uuid)</code></pre>
```
1. typo: kafka-stor[e]ge.sh -> kafka-storage.sh
2. in shell, variable assignment should has no space between "=", i.e.
`KAFKA_CLUSTER_ID=$(bin/kafka-storage.sh random-uuid)`

jsancio replied:
Done.

showuon reviewed (commented):
@jsancio , thanks for the PR, left some comments.

jsancio committed:
Fix quickstart commands

Fix typo in kafka-storage command and add kraft combined mode log to the
remove files command.

jsancio commented:
Thanks for the review @showuon . I addressed your comments. I also fixed the `rm` command at the end of the quickstart.

jsancio committed:
Update the Stream Quickstart to support KRaft

cmccabe commented on docs/streams/quickstart.html:
```diff
@@ -98,19 +97,46 @@ <h4><a id="quickstart_streams_download" href="#quickstart_streams_download">Step
 
 <p>
-Kafka uses <a href="https://zookeeper.apache.org/">ZooKeeper</a> so you need to first start a ZooKeeper server if you don't already have one. You can use the convenience script packaged with kafka to get a quick-and-dirty single-node ZooKeeper instance.
+  Apache Kafka can be started using ZooKeeper or KRaft. To get started with either configuration follow one the section below but not both.
```
should be "one of the sections below" not "one the section"

cmccabe reviewed (approved):
LGTM, left one comment

forlack reviewed (commented):
We should move “Once all services have successfully launched, you will have a basic Kafka environment running and ready to use.” under both ZK and KRaft sections.

jsancio committed:
Fix typo and reaffirm that the user has a successful Kafka cluster.

jsancio commented:
> We should move “Once all services have successfully launched, you will have a basic Kafka environment running and ready to use.” under both ZK and KRaft sections.

@forlack Thanks for the feedback. Kept the sentence in the "Kafka with ZooKeeper" section and added a similar sentence to the "Kafka with KRaft" section. It is awkward formatting to have a paragraph for a parent section after starting a subsection.
